src/providers/v3/gas-data-provider.ts

```diff
@@ -121,9 +121,10 @@ export class ArbitrumGasDataProvider
       this.provider
     );
     const gasData = await gasDataContract.getPricesInWei();
+    const perL1CalldataByte = gasData[1];
     return {
       perL2TxFee: gasData[0],
-      perL1CalldataFee: gasData[1],
+      perL1CalldataFee: perL1CalldataByte.div(16),
```

Comment:
This is great catch! I took a look at [OVM_GasPriceOracle](https://github.com/ethereum-optimism/optimism/blob/7fcdf3daa60272012d4f4c4d884c89b14b0411be/packages/contracts-bedrock/src/L2/GasPriceOracle.sol#L86), where we borrow the [code](https://github.com/Uniswap/smart-order-router/blob/1c93e133c46af545f8a3d8af7fca3f1f2dcf597d/src/util/gas-factory-helpers.ts#L217) to calculate L1 gas. Indeed the result has the factor of 16 multiplied.

You might notice that for Arbitrum gas estimate, smart-order-router should use the arbitrum L1 gas estimate, instead of optimism one. As a follow-up, Uniswap internally probably will need to use the [gasEstimateL1Component](https://github.com/OffchainLabs/nitro-contracts/blob/main/src/node-interface/NodeInterface.sol#L112), assuming this is pretty accurate estimate of the calldata costs?

Reply:
that’s correct for a more accurate estimate but would require an extra rpc call

Reply:
I thought [gasEstimateL1Component](https://github.com/OffchainLabs/nitro-contracts/blob/main/src/node-interface/NodeInterface.sol#L112) returns both L1 gas estimate, as well as L2 base fee?

Reply:
gasEstimateL1Component returns 3 things
1. gasEstimateForL1: an estimate of the amount of gas needed for the l1 component of this tx
2. l2BaseFee: the l2 base fee
3. l1BaseFeeEstimate: ArbOS's l1 estimate of the l1 base fee

gasEstimateForL1 is how much `L2 gas` we charge for the L1 calldata cost
so if you want to calculate feeForL1, that would be `gasEstimateForL1 * l2BaseFee`